### PR TITLE
Remove Blockchain classes

### LINKED ISSUE #178: Implement address(string: String) -> Address?
In order to remove `override public func address(string: String) -> Address?` on blockchain class on iOS, we need to implement this functionality

### PR DESCRIPTION
**Pull request by hewigovens**
## Description

Fix #178 

## Testing instructions

`./bootstrap.sh` and `tools/ios-test`

## Types of changes

Breaking changes:

1. derive address from coin and public key
2. remove blockchain class
3. fix Zcash derivation test
4. add generated files

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: swift/Tests/PrivateKeyTests.swift [hunk 1]
```diff
@@ -32,4 +32,15 @@ class PrivateKeyTests: XCTestCase {
 
         XCTAssertEqual(publicKey.data.hexString, "0499c6f51ad6f98c9c583f8e92bb7758ab2ca9a04110c0a1126ec43e5453d196c166b489a4b7c491e7688e6ebea3a71fc3a1a48d60f98d5ce84c93b65e423fde91")
     }
+
+    func testPrivateKeyToWIF() {
+        // taken from https://en.bitcoin.it/wiki/Wallet_import_format
+        let wif = "5HueCGU8rMjxEXxiPuD5BDku4MkFqeZyd4dZ1jvhTVqvbTLvyTJ"
+        let data = Data(hexString: "0x0C28FCA386C7A227600B2FE50B7CAE11EC86D3BF1FBE471BE89827E19D72AA1D")!
+        let key = PrivateKey(data: data)!
```
Comment (vikmeup) [inline swift/Tests/PrivateKeyTests.swift]: are these tests covered in C++ too?

File: src/Coin.cpp [hunk 2]
```diff
@@ -327,22 +365,67 @@ DerivationPath TW::derivationPath(TWCoinType coin) {
...
+}
+
+std::string TW::deriveAddress(TWCoinType coin, const PublicKey& publicKey) {
     switch (coin) {
     case TWCoinTypeBinance:
-        return Tendermint::Address(HRP_BINANCE_TEST, privateKey.getPublicKey(PublicKeyType::secp256k1)).string();
-    
+        return Tendermint::Address(HRP_BINANCE_TEST, publicKey).string();
+
     case TWCoinTypeCosmos:
```
Comment (vikmeup) [inline src/Coin.cpp]: 👍

File: swift/Sources/Blockchains/BinanceChain.swift [hunk 3]
```diff
@@ -6,35 +6,10 @@
...
-    }
-}
-
-public class BinanceChainTestnet: BinanceChain {
-    override public var hrp: HRP {
-        return .binanceTest
-    }
+struct BinanceChain {
+    static let chainId = "Binance-Chain-Nile"
+    static let source: Int64 = 2
```
Comment (vikmeup) [inline swift/Sources/Blockchains/BinanceChain.swift]: `BinanceChain ` should we move this to C++ code? as static variable

File: swift/Sources/Blockchains/Bitcoin.swift [hunk 4]
```diff
@@ -12,141 +12,21 @@ public typealias TWBitcoinUnspentTransaction = TW_Bitcoin_Proto_UnspentTransacti
...
+public extension HDWallet {
+    static func derive(from extended: String, at path: DerivationPath) -> PublicKey? {
+        let xpubVersion: HDVersion
+        let xprvVersion: HDVersion
+        switch path.purpose {
         case .bip44:
-            let hash = Data([p2pkhPrefix]) + publicKey.bitcoinKeyHash
-            return BitcoinAddress(data: hash)!
+            xpubVersion = path.coinType == .litecoin ? .ltub: .xpub
+            xprvVersion = path.coinType == .litecoin ? .ltpv: .xprv
```
Comment (vikmeup) [inline swift/Sources/Blockchains/Bitcoin.swift]: can you add a test for this one? it is a little tricky logic

File: swift/Sources/Blockchains/Stellar.swift [hunk 5]
```diff
@@ -12,32 +12,3 @@ public typealias TWStellarMemoText = TW_Stellar_Proto_MemoText
 public typealias TWStellarMemoVoid = TW_Stellar_Proto_MemoVoid
```
Comment (vikmeup) [inline swift/Sources/Blockchains/Stellar.swift]: do we still needed `typealias` ? can we generate them or have in one class?

File: swift/Tests/Blockchains/TronTests.swift [hunk 6]
```diff
@@ -10,10 +10,7 @@ import TrustWalletCore
 class TronTests: XCTestCase {
 
     func testTronAddress() {
-        let blockchain = Tron()
-        let address = blockchain.address(string: "TLWEciM1CjP5fJqM2r9wymAidkkYtTU5k3")
-
-        XCTAssertEqual(address?.description, "TLWEciM1CjP5fJqM2r9wymAidkkYtTU5k3")
-        XCTAssertTrue(address is TronAddress)
```
Comment (vikmeup) [inline swift/Tests/Blockchains/TronTests.swift]: probably don't even need this test anymore

File: swift/Sources/Blockchains/Bitcoin.swift (hunk 4 above)
Comment (hewigovens) [reply swift/Sources/Blockchains/Bitcoin.swift]: Yes, we have ltc tests to cover this

File: swift/Sources/Blockchains/Stellar.swift (hunk 5 above)
Comment (hewigovens) [reply swift/Sources/Blockchains/Stellar.swift]: Yes, there is an issue for that

File: swift/Tests/Blockchains/TronTests.swift (hunk 6 above)
Comment (hewigovens) [reply swift/Tests/Blockchains/TronTests.swift]: iOS still needs it, we should export TWTronAddress later

File: swift/Sources/Blockchains/BinanceChain.swift (hunk 3 above)
Comment (hewigovens) [reply swift/Sources/Blockchains/BinanceChain.swift]: Will do

File: swift/Tests/PrivateKeyTests.swift (hunk 1 above)
Comment (hewigovens) [reply swift/Tests/PrivateKeyTests.swift]: I was considering remove `wif` things, not that useful, @alejandro-isaza do you agree?

Comment (alejandro-isaza) [reply swift/Tests/PrivateKeyTests.swift]: Yes, I don't think we use it.

**Pull request merged**